build-system/externs/amp.extern.js

```diff
@@ -223,7 +241,18 @@ window.__AMP_TOP;
 window.__AMP_PARENT;
 window.__AMP_WEAKREF_ID;
 window.__AMP_URL_CACHE;
-window.AMP = {};
+window.__AMP_LOG;
+
+/** @type {boolean} */
+window.ENABLE_LOG;
+
+/** @type {../../src/mode.ModeDef} */
+window.__AMP_MODE;
```

Comment:
This and above  are `{undefined|blahblah}`
This is an example of the kind of incorrect typing that could be masked if we're not careful, causing the typechecking to allow uses of some of these values before they're initialized and without checking first.

Reply:
1. Unless a `!` is provided, you'll still need to check against `null`, since this has an implied possibly null as-is.
2. I'm fully onboard with doing a complete sweep of `amp.extern.js` and making wholesale improvements. Upstreaming types to closure, organizing the code, and improving imprecise types.

Reply:
I've applied the fixes